CDCgov/prime-reportstream
# 2885 Load lookup tables locally for testing

This PR adds functionality to the CLI to load a series of tables from a folder and activate the tables to make it easy to load tables in our local environments.  It also adds the automated loading of the tables when starting the local container.  Note that this new load command is not allowed in production, just so there are no mistakes.

Test Steps:
Currently there are no tables being stored in the database, so we will use some of the existing ones for this test.
1. Copy two or more tables from metadata/tables to src/test/resources/metadata/tables.  Rename the extensions from .tables to .csv.
2. Run `gradle package` and run your functions.
3. If you run the functions in the docker container then verify the tables were loaded automatically by running `../prime lookuptables list` and verifying the tables you added are listed.
4. Run the CLI to load add the tables: `./prime lookuptables loadall`
5. Verify the tables were loaded by getting the list of tables with `./prime lookuptables list`.  If your container loaded the tables already then verify a new version was added.

## Changes
- CLI can load a full folder of tables
- Tables are loaded during the start of the local container
- Added gradle task to load the tables
- Changed gradle task reloadDb to resetDb to try to make it obvious what it does
- Changed the table creation to batched which made large table uploads.  NCIS table (12200+ rows) went from 5 mins to 25 secs

## Checklist

### Testing

### Process

## Fixes
- #issue - List GitHub issues this PR fixes

## To Be Done
*Create GitHub issues to track the work remaining, if any*
- #issue

## Specific Security-related subjects a reviewer should pay specific attention to
- Does this PR introduce new endpoints?
    - new endpoint A
    - new endpoint B
- Does this PR include changes in authentication and/or authorization of existing endpoints?
- Does this change introduce new dependencies that need vetting?
- Does this change require changes to our infrastructure?
- Does logging contain sensitive data?
- Does this PR include or remove any sensitive information itself?

If you answered '_yes_' to any of the questions above, conduct a detailed Review that addresses at least:

- What are the potential security threats and mitigations? Please list the _STRIDE_ threats and how they are mitigated
    - **S**poofing (faking authenticity)
        - Threat _T_, which could be achieved by _A_, is mitigated by _M_
    - **T**ampering (influence or sabotage the integrity of information, data, or system)
    - **R**epudiation (the ability to dispute the origin or originator of an action)
    - **I**nformation disclosure (data made available to entities who should not have it)
    - **D**enial of service (make a resource unavailable)
    - **E**levation of Privilege (reduce restrictions that apply or gain privileges one should not have)
- Have you ensured logging does not contain sensitive data?
- Have you received any additional approvals needed for this change?
